frame.go

```diff
@@ -404,24 +405,58 @@ func newAckFrame(rs ackRanges) (*frame, error) {
 	f.NumTS = 0
 	f.TimestampSection = nil
 
-	for i := 1; i < len(rs); i++ {
-		gap := last - rs[i].lastPacket
-		assert(gap < 256) // TODO(ekr@rtfm.com): handle this.
+	addedRanges := 1
+
+	/* SECOND, add the remaining ACK blocks that fit and that we have */
+	for (maxackblocks > f.NumBlocks) && (addedRanges < len(rs)) {
+
+		/* calculate blocks needed for the next range */
+		gap := last - rs[addedRanges].lastPacket - 1
+		blocksneeded := uint64((gap / maxAckGap) + 1)
```

Comment:
Hmmm. What happens if gap is an even multiple of maxAckGap

Reply:
I think you're on to a bug here, should probably be something like `uint64(ceil(gap / maxAckGap))`

Reply:
Hmm.... I think go arithmetic is integer. I think you want:
```
(gap + (maxAckGap - 1)) / maxAckGap
```

Reply:
Exactly.